CIP-0057/README.md

```diff
@@ -0,0 +1,414 @@
+
+For all such properties, validation succeeds if the child instance validates against the "additionalProperties" schema. Omitting this keyword has the same assertion behavior as an empty schema.
+
+##### propertyNames
+
+The value of "propertyNames" must be a valid JSON Schema.
+
+If the instance is an object, this keyword validates if every property name in the instance validates against the provided schema. Note the property name that the schema is testing will always be a string. Omitting this keyword has the same behavior as an empty schema.
+
+##### maxProperties
+
+The value of this keyword must be a non-negative integer. An object instance is valid against "maxProperties" if its number of properties is less than, or equal to, the value of this keyword.
+
+##### minProperties
+
+The value of this keyword must be a non-negative integer. An object instance is valid against "maxProperties" if its number of properties is greater than, or equal to, the value of this keyword.
+
+#### For `{ "dataType": "constructor" }`
+
+> 💡 Keywords in this section only applies to `constructor`. Using them in conjunction with an invalid data-type should result in an error.
+
+##### index
+
+This keyword's value must be a non-negative integer. An instance is valid against this keyword if it represents a Plutus constructor whose index is the same as this keyword's value. 
+
+##### fields
+
+This keyword's value must be an array of valid _Plutus Data Schema_; possibly empty. It must also always be specified with the `index` keyword. An instance is valid against this keyword if it represents a Plutus constructor for which each field is valid under each subschema given by this keyword's value. Fields are compare positionally. 
+
+## Example(s)
+
+<details>
+  <summary>Lobster Challenge Contract</summary>
+
+```json
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+
+  "$id": "https://cips.cardano.org/cips/cip57/schemas/v1/lobster-challenge.json",
+
+  "$vocabulary": {
+    "https://json-schema.org/draft/2020-12/vocab/core": true,
+    "https://json-schema.org/draft/2020-12/vocab/applicator": true,
+    "https://json-schema.org/draft/2020-12/vocab/validation": true,
+    "https://cips.cardano.org/cips/cip57": true
+  },
+
+  "preamble": {
+    "title": "Lobster Challenge",
+    "description": "A specification of the lobster challenge contract: <https://github.com/input-output-hk/lobster-challenge/>",
+    "version": "1.0.0",
+    "license": "Apache-2.0"
+  },
+  "validators": {
+    "main": {
+      "description": "The lobster contract validator, verifying vote counts and driving token evolution.",
+      "datum": {
+        { "$ref": "plutus-types.json#/$defs/Unit" }
+      },
+      "redeemer": {
+        { "$ref": "plutus-types.json#/$defs/Unit" }
+      },
+      "compiledCode": {
+        "template": "59094c01000{{ lobsterParams }}006002003",
```

Comment:
What does this mean?

Reply:
Read the specification: https://mustache.github.io/mustache.5.html

Although, this part -- the templating / compiled code -- is arguably clunky. There's just no good way today to obtain a parameterized contract code than to recompile it; I'll probably end up removing this one from the proposal in the end.

Reply:
I realize the only mention to mustache is in the meta-schema; which is a bit like -- hidden far away. If kept, this should be mentioned more clearly in the main specification document itself.